### PR TITLE
Feat/open ai compatible functioncall

### PR DESCRIPTION
**Pull request by JohnJyong**
# Description
Open AI compatible support function call

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by takatost — APPROVED** [review]
LGTM

File: api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py [hunk 1]
```diff
@@ -166,11 +167,23 @@ def get_customizable_model_schema(self, model: str, credentials: dict) -> AIMode
...
             generate custom model entities from credentials
         """
+        support_function_call = False
+        features = []
+        function_calling_type = credentials.get('function_calling_type', 'no_call')
+        if function_calling_type == 'function_call':
+            features = [ModelFeature.TOOL_CALL]
+            support_function_call = True
+        endpoint_url = credentials["endpoint_url"]
+        # if not endpoint_url.endswith('/'):
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py:177]: Please remove comments

File: api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py [hunk 2]
```diff
@@ -522,33 +541,34 @@ def _convert_prompt_message_to_dict(self, message: PromptMessage) -> dict:
...
             if message.tool_calls:
-                message_dict["tool_calls"] = [helper.dump_model(PromptMessageFunction(function=tool_call)) for tool_call
-                                              in
-                                              message.tool_calls]
-                # function_call = message.tool_calls[0]
-                # message_dict["function_call"] = {
-                #     "name": function_call.function.name,
-                #     "arguments": function_call.function.arguments,
-                # }
+                # message_dict["tool_calls"] = [helper.dump_model(PromptMessageFunction(function=tool_call)) for tool_call
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py:544]: Please remove comments

File: api/core/model_runtime/model_providers/openai_api_compatible/openai_api_compatible.yaml [hunk 3]
```diff
@@ -75,6 +75,28 @@ model_credential_schema:
...
+        en_US: Function calling
+      type: select
+      required: false
+      default: no_call
+      options:
+        - value: function_call
+          label:
+            en_US: Support
+            zh_Hans: 支持
+#        - value: tool_call
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/openai_api_compatible/openai_api_compatible.yaml:92]: Please remove comments

**Pull request merged**